@xen-orchestra/backups/writers/DeltaBackupWriter.js

```diff
@@ -40,7 +40,8 @@ exports.DeltaBackupWriter = class DeltaBackupWriter extends MixinBackupWriter(Ab
         await asyncMap(vhds, async path => {
           try {
             await checkVhdChain(handler, path)
-            found = found || (await adapter.isMergeableParent(packedBaseUuid, path))
+            const isMergeable = await adapter.isMergeableParent(packedBaseUuid, path)
+            found = found || isMergeable
```

Comment:
Congrats for finding this bug @fbeauchamp :clap: :trophy: 

Let's avoid calling `isMergeableParent` if not necessary:
```suggestion
            if (!found) {
              const isMergeable = await adapter.isMergeableParent(packedBaseUuid, path)
              
              // found might have been set to true concurrently, make sure to not erase it
              found ||= isMergeable
            }
```

Reply:
Either we do this sequentially and we can stop it as soon as the vhd is  found OR we let this in parallel and isMergeableParent will be called for each vhd 

But your proposition  won't limit the call to isMergeableParent  (as long as we talk of a a few dozen vhd max). 

For example this code 
```js 
function timeout(seconds){
  return new Promise((resolve, reject)=>{
    setTimeout(()=>resolve(true), seconds * 1000)
  })
}
let found = false
async function test(){
  const numbers = []
  for(let i=0; i < 10 ; i ++){
    numbers.push(i)
  }
  await Promise.all(
    numbers.map(async i=>{
      console.log('start ', i )
      found = found || await timeout(i)
      console.log('done ', i, res)
    })
  )

}
```
will lead to start 0 -> start 9 and then done 0 to done 1

Reply:
Indeed, if the concurrency is not limited, it will not change anything :slightly_smiling_face:

Reply:
At least a comment explaining the reasoning of this change appears necessary.